### PR TITLE
Ir sensor

### PR DESCRIPTION
**Pull request by ltchin**

### CODE REVIEW

File: analog_read.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 class AnalogRead(SyncedSketch):
 
-    adc_pin = 0
```
Comment (eric-wieser) [inline analog_read.py:8]: Probably don't want this

File: tamproxy/comm/tamp_serial.py [hunk 2]
```diff
@@ -44,7 +44,8 @@ def get_port(self):
         else:
             print "Enter the index of the desired serial port:"
             for i,p in enumerate(ports): print i,p
-            return ports[int(raw_input())]
+            return ports[0]
+            # return ports[int(raw_input())]
```
Comment (eric-wieser) [inline tamproxy/comm/tamp_serial.py]: This should probably go upstream, but not in this changeset - we've hit this problem elsewhere

File: tamproxy/devices/long_ir.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    def add_payload(self):
+        return self.DEVICE_CODE + chr(self.pin)
+
+    def _handle_update(self, request, response):
+        self.distTicks = (ord(response[0])<<8) + ord(response[1])
+        # Set cutoff of where can't really tell values well
+        if self.distTicks < self.TOO_FAR_TICKS:
+            self.distInches = "far"
+        elif self.distTicks > self.DISCONNECTED:
+            self.distInches = "disconnected"
```
Comment (eric-wieser) [inline tamproxy/devices/long_ir.py]: throw ValueError / IOError instead?

File: tamproxy/devices/long_ir.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    @property
+    def add_payload(self):
+        return self.DEVICE_CODE + chr(self.pin)
+
+    def _handle_update(self, request, response):
+        self.distTicks = (ord(response[0])<<8) + ord(response[1])
+        # Set cutoff of where can't really tell values well
+        if self.distTicks < self.TOO_FAR_TICKS:
+            self.distInches = "far"
```
Comment (eric-wieser) [inline tamproxy/devices/long_ir.py]: float('inf') might be more useful

File: tamproxy/devices/ultrashort_ir.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+    DEVICE_CODE =   c.devices.digital_input.code
+    READ_CODE =     c.devices.digital_input.read_code
+
+    def __init__(self, tamproxy, pin, pullup=True, continuous=False):
+        self.pin = pin
+        self.pullup = pullup
+        self.val = 0
+        self.prev_val = None
+        super(DigitalInput, self).__init__(tamproxy, continuous)
```
Comment (eric-wieser) [inline tamproxy/devices/ultrashort_ir.py]: digitalinput will need the pin in its constructor

File: tamproxy/comm/tamp_serial.py (hunk 2 above)
Comment (ltchin) [inline tamproxy/comm/tamp_serial.py]: Oh oops, this is a cheese for my computer. Forgot to get rid of it

File: tamproxy/devices/long_ir.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import warnings
+from .device import ContinuousReadDevice
+from .. import config as c
+
+class LongIR(ContinuousReadDevice):
+
+    DEVICE_CODE =   c.devices.ir.code
+    READ_CODE =     c.devices.ir.code
```
Comment (eric-wieser) [inline tamproxy/devices/long_ir.py]: `.read_code`?

**Pull request merged**